src/framework/components/element/pointer-events.js

```diff
@@ -0,0 +1,190 @@
+        },
+
+        // Mouse-specific event handler.
+        _onMouseDown: function(mouseEvent) {
+            this._pointerEventDown( new pc.Vec3( mouseEvent.x, mouseEvent.y, 0 ) );
+        },
+
+        // Mouse-specific event handler.
+        _onMouseUp: function(mouseEvent) {
+            this._pointerEventUp( new pc.Vec3( mouseEvent.x, mouseEvent.y, 0 ) );
+        },
+
+        // Mouse-specific event handler.
+        _onMouseMove: function(mouseEvent) {
+            this._pointerEventMove( new pc.Vec3( mouseEvent.x, mouseEvent.y, 0 ) );
+        },
+
+        // Mouse-specific event handler.
+        _onMouseMove: function(mouseEvent) {
+            this._pointerEventMove( new pc.Vec3( mouseEvent.x, mouseEvent.y, 0 ) );
+        },
+
+        // Touch-specific event handler.
+        _onMouseWheel: function(mouseEvent) {
+            this._pointerEventScroll( new pc.Vec3( mouseEvent.x, mouseEvent.y, 0 ), mouseEvent.wheel );
+        },
+
+        // Touch-specific event handler.
+        _onTouchUp: function(touchEvent) {
+            var touch = touchEvent.changedTouches[0];
+            this._pointerEventUp( new pc.Vec3( touch.x, touch.y, 0 ) );
+        },
+
+        // Touch-specific event handler.
+        _onTouchMove: function(touchEvent) {
+            var touch = touchEvent.changedTouches[0];
+            this._pointerEventMove( new pc.Vec3( touch.x, touch.y, 0 ) );
+        },
+
+        /**
+        * @function
+        * @name pc.ScreenComponent#enablePointerEvents
+        * @description Starts listening to mouse and touch events for the currenct {@link pc.ScreenComponent} instance.
+        * @example
+        * // On an entity with a screen component
+        * entity.screen.enablePointerEvents();
+        */
+        enablePointerEvents: function(_app) {
+            var app = _app || pc.Application.getApplication();
+
+            if (app.mouse) {
+                app.mouse.on(pc.EVENT_MOUSEDOWN,   this._onMouseDown,   this);
+                app.mouse.on(pc.EVENT_MOUSEUP,     this._onMouseUp,     this);
+                app.mouse.on(pc.EVENT_MOUSEMOVE,   this._onMouseMove,   this);
+                app.mouse.on(pc.EVENT_MOUSEWHEEL,  this._onMouseWheel,  this);
+            }
+
+            if (app.touch) {
+                app.touch.on(pc.EVENT_TOUCHSTART,  this._onTouchDown,   this);
+                app.touch.on(pc.EVENT_TOUCHEND,    this._onTouchUp,     this);
+                app.touch.on(pc.EVENT_TOUCHMOVE,   this._onTouchMove,   this);
+                app.touch.on(pc.EVENT_TOUCHCANCEL, this._onTouchUp,     this);
+            }
+
+            this._rootPointerEventReceiver = true;
+        }
+
+    };
+
+    // we mix point events in to both Screen and Element so that the whole UI stack
+    // becomes mouse/touch events-aware.
+    pc.extend(pc.ScreenComponent.prototype, PointEventsManager);
+    pc.extend(pc.ElementComponent.prototype, PointEventsManager);
+
+    return {
```

Comment:
I've worked with quiet few mouse/touch interactions, and did came up with thing I've called `taps`, where developer works with abstraction where he does not know if it is mouse or touch, and there is only this list of events:
`down`
`click`
`dragstart`
`dragmove`
`dragend`

Where if mouse/touch where not moved too far from initial down location and did get released within something like 300ms, then it will be click event. But if mouse/touch stayed at same location 300ms or moved few pixels, then it will be dragstart, for every next move dragmove and on release dragend. This provides unified touch/mouse UX and allows developer to do pretty much any interaction required. For mouse there were additional things provided on event, such as button.